wallet/src/ui/app/hooks/useMutationObserver.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+import type { RefObject } from 'react';
+
+interface Props {
+    target: RefObject<Element> | Element | Node | null;
```

Comment:
Why not just make this support one of these instead of all of these? Seems like it'd get rid of `getRefElement` entirely.